### ISSUE TITLE
Problems with new feature - Improve demangled names translation

### ISSUE THREAD

**Issue by user202729** [igogo-x86/HexRaysPyTools#20]
About the commit https://github.com/igogo-x86/HexRaysPyTools/commit/7668c2d80cabe9205a6a96c016b761fedc672eb5

* There is a typo in the operator sub, mul and div.
* In the current state, operator ptr and operator mul cannot be distinguished.
* `operator<` is missing, as well as some other comparison operators.
* Instead of chained `elif` list, I suggest doing something like this

```
name = name.replace("operator==", "operator_EQ_")
name = name.replace("operator!=", "operator_NEQ_")
# ...
```

That way it's also possible to avoid false detection (if identifier name contains `operator` as a substring), and don't (unnecessarily) error out if there is a `operator XXX`.

**Comment by igogo-x86** [comment]
Yeah, I've already stumbled upon several problems with this translator. Now I fixed all types and suppose added all possible operators. 

Indeed there's no simple way to tell pointer from multiplication so let it be "_STAR_" for now. As for using `replace` method instead of bunch `elif`s - process of analysis all function names is started at the moment of opening database and I considered performance issues. But I agree that your method is less error-prone.

**Comment by user202729** [comment]
There are some `[idx:idx]` error (which results in the empty string) at the [new commit](https://github.com/igogo-x86/HexRaysPyTools/commit/66014e87c8f83c2f97bb856c00630f0b5aec6044).

Also, won't this cause an error with a function name such as `fn_1234_operator`?

If you're worried about performance, you can try using a dict (hash table) instead:

```
try:
	idx = name.rindex('operator')+8
	name = name[:idx]+{
		'<<=': '_LEFT_SHIFT_ASSIGN_',
		# etc.
	}[name[idx:]]
except:
	pass
```

Dict lookup should be faster than elif chain because the latter compares the strings one-by-one.

**Comment by igogo-x86** [comment]
Idea with a dictionary is quite good but unfortunately name doesn't always end after `operator[]`. It can be for example like this:
`public: z3::model::operator[](int)const`

For functions with word `operator` as part of their name I've added a check if symbol after `operator` is exist and forbidden.

Actually, I've found one more problem with `public`, `private` and `protected` specifiers. They have single `:` symbol and it should be replaced either way virtual tables wouldn't be created. Every method for reverse-engineer is public  so they are just ignored in process of translation now.
